storage-proofs/core/src/gadgets/insertion.rs

```diff
@@ -21,6 +21,23 @@ pub fn insert<E: Engine, CS: ConstraintSystem<E>>(
     let size = elements.len() + 1;
     assert_eq!(1 << bits.len(), size);
 
+    match size {
```

Comment:
Maybe just write this as
```rust
if size == 4 {
    return ...
} else if size == 8 {
    return ...
}
```
to avoid the empty match arm and the `if`s in pattern matching?

Reply:
Done.